repos/system_upgrade/common/libraries/module.py

```diff
@@ -23,14 +23,16 @@ def _create_or_get_dnf_base(base=None):
         # have repositories only for the exact system version (including the minor number). In a case when
         # /etc/yum/vars/releasever is present, read its contents so that we can access repositores on such systems.
         conf = dnf.conf.Conf()
-        pkg_manager = 'yum' if get_source_major_version() == '7' else 'dnf'
-        releasever_path = '/etc/{0}/vars/releasever'.format(pkg_manager)
-        if os.path.exists(releasever_path):
+
+        # dnf on EL7 doesn't load vars from /etc/yum, so we need to help it a bit
+        releasever_path = '/etc/yum/vars/releasever'
+        if get_source_major_version() == '7' and os.path.exists(releasever_path):
             with open(releasever_path) as releasever_file:
```

Comment:
I would put the releasever_path setting inside of the condition to show that it is only a part of the code needed to handle EL7:
```suggestion
        # dnf on EL7 doesn't load vars from /etc/yum, so we need to help it a bit
        if get_source_major_version() == '7' and os.path.exists(releasever_path):
            releasever_path = '/etc/yum/vars/releasever'
            with open(releasever_path) as releasever_file:
```

A general note: using a `try: except:` around `open()` and passing if the file does not exist is resistant to race conditions whereas checking file existence first and then operating on the file is not so I generally prefer the `try: except:` strategy.  However, I see (1) that we use the file existence check strategy all over the code right now and (2) this specific case is unlikely to have the file deleted out from under us. So, something to keep in mind but I won't require that change to be made.

Reply:
I can't call `os.path.exists(releasever_path)` before assigning `releasever_path = '/etc/yum/vars/releasever'` ;-)

The logic I wanted to keep is "if on EL7 *and* the file exists, load it, in *all* other cases use `get_source_major_version`.

I could rewrite this to:
```
conf.substitutions['releasever'] = get_source_major_version()
if get_source_major_version() == '7':
  releasever_path = '/etc/yum/vars/releasever'
  if os.path.exists(releasever_path):
    with …:
      conf.substitutions['releasever'] = releasever_file.read().strip()
```

But it doesn't make the code much more readable, IMHO.

Reply:
Or even
```python
conf.substitutions['releasever'] = get_source_major_version()

if get_source_major_version() == '7':
  try:
    with open('/etc/yum/vars/releasever') as releasever_file:
      conf.substitutions['releasever'] = releasever_file.read().strip()
  except:
    pass

# load all substitutions from etc
conf.substitutions.update_from_etc('/')
```

Reply:
Done this now. Let me know what you think.

Reply:
+1

Reply:
And yeah, I got ahead of myself thinking about both changing to `try: except:` and moving the assignment but then only making one of those a suggestion.